docs/gen-versions.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python3
+
+import argparse
+import json
+import os
+import subprocess
+import sys
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        prog="gen-versions.py",
+        description="Generate Sphinx version switcher JSON file from git tags.",
+    )
+
+    parser.add_argument(
+        "-o",
+        "--out-file",
+        help="path to output file, including filename, for generated versions JSON file",
+        metavar="path",
+        default=None,
+    )
+
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+
+    completed = subprocess.run(["./gen-versions.sh"], capture_output=True)
+    versions = completed.stdout.splitlines()
+
+    output = []
+
+    # Special case for latest.
+    latest = versions.pop(0)
+    latest = latest.decode()
+    output.append(
+        {
+            "version": latest,
+            "url": f"https://docs.determined.ai/latest/",
```

Comment:
shouldnt this work with both urls? or is that not possible?

Reply:
maybe just don't include the `.pop`?

Reply:
im guessing you're preserving functionality here, just wanted to throw comment

Reply:
I'm going to step through this again and refresh my memory. 😅

Reply:
At the very least, it doesn't need to be an "f-string" since there are no interpolated args.
```suggestion
            "url": "https://docs.determined.ai/latest/",
```

Reply:
I stepped through everything again, and this is all preserving functionality, although I believe Sphinx cares about the ordering of the array items for `versions.json` (see the [current production version](https://docs.determined.ai/latest/_static/version-switcher/versions.json)). The called shell script returns versions in descending order, so I remove the first list element for the special case of `latest`, then continue with the remaining versions.

Reply:
I've removed the spurious f-string.